scheduler/src/main/java/org/pentaho/platform/scheduler2/quartz/EmbeddedQuartzSystemListener.java

```diff
@@ -112,7 +114,18 @@ public boolean startup( final IPentahoSession session ) {
         if ( logger.isDebugEnabled() ) {
           logger.debug( scheduler.getQuartzScheduler().getSchedulerName() );
         }
-        scheduler.start();
+
+        File schedulerStatusFile = new File(
+          PentahoSystem.getApplicationContext().getSolutionPath( "system" + File.separator + "quartz" ) + File.separator
+            + "scheduler.status" );
+
+        // If the scheduler status file exists and contains "RUNNING" status, start the scheduler
+        if ( schedulerStatusFile.exists() && FileUtils.readFileToString( schedulerStatusFile, StandardCharsets.UTF_8 )
+          .equals( IScheduler.SchedulerStatus.RUNNING.name() ) ) {
+          scheduler.start();
+        } else {
+          scheduler.pause();
+        }
```

Comment:
I think it would be best to have both the load and store logic in the same class.

You could add a method such as `getStoredStatus()` to the `QuartzScheduler` class and use that to decide whether to start the service.

Alternatively, you could add a method such as `restoreStoredStatus()` to the `QuartzScheduler` class that would handle reading the stored state and possibly start the service.